flink-dist/src/main/flink-bin/bin/config.sh

```diff
@@ -296,7 +296,10 @@ bin=`dirname "$target"`
 SYMLINK_RESOLVED_BIN=`cd "$bin"; pwd -P`
 
 # Define the main directory of the flink installation
-FLINK_HOME=`dirname "$SYMLINK_RESOLVED_BIN"`
+# If config.sh is called by pyflink-shell.sh in python bin directory(pip installed), use the _PYFLINK_HOME
+if [ -z "$_FLINK_HOME_DETERMINED" ]; then
```

Comment:
In case that `FLINK_HOME` is already determined, `FLINK_HOME` will be set. So we can check if `FLINK_HOME` is already set and there is no need to add variable `_FLINK_HOME_DETERMINED`. What do you think?

Reply:
IMO we should not check FLINK_HOME is already set. Because user could also set FLINK_HOME as a environment variable. But the original behavior of config.sh is to set its own directory to FLINK_HOME, regardless of whether the user has set FLINK_HOME. So if we just use the FLINK_HOME in environment, the behavior of config.sh will be completely changed. 
This is not our intention.